### ISSUE TITLE
Эксперименты с гиперпараметрами модели

### ISSUE THREAD

**Issue by tatyanamarega** [tatyanamarega/Supervised_Learning_Real#2]
Результаты эксперимента (Issue #1)

Сравнение моделей:
| Модель                        | Accuracy | Ключевые параметры                          |
|--------------------------|-----------|--------------------------------------------|
| **RandomForest**       | 0.8923    | n_estimators=500, max_depth=20         |
| SVM                             | 0.8891    | C=0.1, kernel=rbf                                   |
| GradientBoosting        | 0.8849    | n_estimators=100, learning_rate=0.2     |
| Logistic Regression     | 0.8860    | C=0.01, solver=lbfgs                               |
| KNN                            | 0.8807    | n_neighbors=13, weights=distance        |
| Decision Tree              | 0.8494    | max_depth=7, min_samples_split=5       |

Выводы:
1. RandomForest - оптимальный выбор для production:
   - Лучшая точность (0.8923)
   - Устойчивость к переобучению
   - Эффективная работа с категориальными признаками

2. GradientBoosting показал потенциал для улучшения:
   - Возможность увеличения n_estimators до 300-500
   - Перспектива использования early stopping

3. SVM сохраняет значимость:
   - Минимальная разница с RandomForest (-0.0032)
   - Хорошая стабильность результатов

4. Интерпретируемость:
   - Для анализа важности признаков: Decision Tree
   - Для объяснимых моделей: Logistic Regression

Пример оптимальной конфигурации RandomForest
best_model = RandomForestClassifier(
    n_estimators=500,
    max_depth=20,
    min_samples_split=2,
    random_state=42
